compiler/testData/codegen/box/optimizations/kt46921.kt

```diff
@@ -0,0 +1,12 @@
+// TARGET_PLATFORM: JVM
```

Comment:
`TARGET_BACKEND` + regenerate tests pls.

But also consider moving this test to `boxInline`, to test the two-module scenario too

Reply:
I'm not sure it makes sense to test this as multiple modules since the purpose of `inline` here is to generate a specific bytecode pattern, not to test the inliner.